homeassistant/helpers/template.py

```diff
@@ -1380,6 +1381,34 @@ def state_attr(hass: HomeAssistant, entity_id: str, name: str) -> Any:
     return None
 
 
+def is_nominal(hass: HomeAssistant, entity_id: str | State | Iterable | None) -> bool:
+    """Test if an entity has a nominal state."""
+    # Entity is a non-existing object
+    if entity_id is None:
+        return False
+
+    # If entity is a string or object then make it a list first.
+    if isinstance(entity_id, (str, State)):
+        entity_id = [entity_id]
+
+    # Loop through entity items provided.
+    for entity in entity_id:
+        if isinstance(entity, str):
+            # Set value to state if string is an entity_id, otherwise to string itself.
+            value = (
+                getattr(_get_state(hass, entity), "state", None)
+                if "." in entity
+                else entity
+            )
+
+            if value is None or value in [STATE_UNAVAILABLE, STATE_UNKNOWN]:
+                return False
+        elif entity is None or entity.state in [STATE_UNAVAILABLE, STATE_UNKNOWN]:
+            return False
```

Comment:
So if you pass in a list of entities like `['sensor.a', 'sensor.b', 'sensor.c']` it returns `True` only if all of them are nominal and `False` if any are not? That seems strange, its basically the inverse of how these [states are handled in groups by default](https://www.home-assistant.io/integrations/group/#binary-sensor-light-and-switch-groups):
> - The group state is unavailable if all group members are unavailable.
> - Otherwise, the group state is unknown if all group members are unknown or unavailable.

I feel like if you're going to allow a list of items as input this should be inverted. It should return `True` if any of them are a nominal state and `False` only if all of them are not.

But also, do we need to allow `Iterable` as input at all? What's the use case for this?

Reply:
Correct, if any of them are not nominal then it returns False.
One of the use cases for this would be the availability of template sensors. Regularly one then makes the template sensor unavailable if any of the entities it relies on are unavailable. So by allowing a list and only returning true if all of them are nominal makes this easy.
For example, template sensor that adds 2 sensors together
```
state: "{{ int(states('sensor.one'),0) + int(states('sensor.two'),0) }}"
availability: "{{ ['sensor.one', sensor.two'] | is_nominal }}"
```

Reply:
Hm. Ok well that use case does make sense. I don't think the behavior in this situation is particularly intuitive though. I guess the question is do we think people will figure this out and use it? Or just use `select` and `reject` instead, something like this:
```
availability: "{{ ['sensor.one', sensor.two'] | reject('nominal') | count == 0 }}"
```

Reply:
I've put in the documentation that it can be a list and examples. Main thing here with the list is to make it more simple to do this. Personally I can see people a lot more do:
``` availability: "{{ ['sensor.one', sensor.two'] | is_nominal }}" ```
then
```availability: "{{ ['sensor.one', sensor.two'] | reject('nominal') | count == 0 }}"``` 
once they see it can be a list as the latter requires more JINJA2 knowledge and understanding. I know for myself I just started using those things. :-)

Reply:
Ok. I guess now that I get the use case I don't really feel strongly either way. If no others have an issue with it then I'm good with it.

Reply:
Current behavior seems logical and useful to me 👌